### PR TITLE
Add Disqus support

### PR DESCRIPTION
**Pull request by mbernard**
Add support for Disqus

Usage:
Simply set your Disqus shortname in the `appConfig.ts` or in the env var `JAMIFY_DISQUS_SHORTNAME`

### CODE REVIEW

**Comment by styxlab** [comment]
Cool, thanks a lot. Will take a closer look next week!

**Comment by styxlab** [comment]
@all-contributors please add @mbernard for code

File: appConfig.ts [hunk 1]
```diff
@@ -28,6 +28,10 @@ export const commento: boolean = false
 export const commentoUrl: string = 'https://cdn.commento.io'
 //export const commentoUrl: string = 'https://commento.your-blog.com'
 
+// disqus commenting system
+export const disqus: boolean = true
```
Comment (styxlab) [inline appConfig.ts:32]: Could you please switch the default to false?

File: components/Post.tsx [hunk 2]
```diff
@@ -149,7 +152,9 @@ export const Post = ({ cmsData }: PostProps) => {
 
                 {memberSubscriptions && <Subscribe {...{ settings }} />}
 
-                {commento.enable && <Comments {...{ id: post.id, url: commento.url }} />}
+                {disqus.enable && (<DisqusComments {...{ post: post, shortname: disqus.Shortname }} />)}
```
Comment (styxlab) [inline components/Post.tsx:155]: Maybe only pass `post.id` and `post.url` instead of the whole post object?

Comment (styxlab) [reply components/Post.tsx:155]: Question: What does disqus need the `post.url` for, shouldn't `post.id` be sufficient just like in commento?

File: appConfig.ts (hunk 1 above)
Comment (styxlab) [reply appConfig.ts:32]: Discussion (I haven't made up my mind): What do you think about diabling disqus when commento is enabled and vice versa, so only one commenting system is allowed at the same time?

In this case we would also need to change the config to something like
```
type CommentingTypes = 'commento' | 'disqus' | false
export const commenting: CommentingTypes = false
...
The `processEnv` logic would need to be updated too. I am leaning towards this change as I don't see that anyone would like to use both systems in parallel.
```

**Review by styxlab — CHANGES_REQUESTED** [review]
Great job! Have a look at my comments for some minor improvements.

File: appConfig.ts (hunk 1 above)
Comment (mbernard) [reply appConfig.ts:32]: Totally agree that it doesn't make sense to have both active at the same time. However I always find it weird when you have a choice, then a specific setting related to that choice, it kinda makes the usage harder IMO. It think it's easier to have a common way to configure all the settings

```
settingsSectionXXXEnabled: TRUE | FALSE
then all the settings related to that section
```

File: components/Post.tsx (hunk 2 above)
Comment (mbernard) [reply components/Post.tsx:155]: It's the way Disqus works. Conversations are linked to the Id and it uses the URL to discover/suggest other conversations on your site. Personally, I think that 3 parameters is limit as for the readability, but I don't mind changing it if you prefer. It just seemed easier at the time as all those properties were available on the PageOrPost type

File: appConfig.ts (hunk 1 above)
Comment (mbernard) [reply appConfig.ts]: otherwise switched default to false

**Review by styxlab — APPROVED** [review]
Thanks for the discussion. I will merge the PR, but will make some small modifications, such as combining the commenting options. For exclusive choices (like here), I prefer one selection option as I find it simpler to change just one "knob" instead of two. I see value in your approach, when you can combine two or more options.

**Pull request merged**